app/api/src/crud/crud_read_heatmap.py

```diff
@@ -439,37 +430,74 @@ async def read_heatmap(
             )
         return sorted_table, unique
 
-    def do_calculations(self, sorted_table, unique, heatmap_settings):
+    def do_calculations(self, sorted_table, uniques, heatmap_settings):
         # TODO: find a better name for this function
         """
         connect the heatmap core calculations to the heatmap method
         """
         method_map = {
-            "gravity": "gravity",
+            "gravity": "modified_gaussian_per_grid",
             "connectivity": "connectivity",
             "cumulative": "counts",
-            "closest_average": "modified_gaussian_per_grid",
+            "closest_average": "mins",
         }
-        method_name = method_map[heatmap_settings.method]
-        method = heatmap_core.getattr(method_name)
         output = {}
-        for key, heatmap_config in heatmap_settings.heatmap_config:
-            output[key] = method(sorted_table[key], unique[key], heatmap_config)
+        if heatmap_settings.heatmap_type.value == "gravity":
+            for key, heatmap_config in heatmap_settings.heatmap_config.items():
+                output[key] = heatmap_core.modified_gaussian_per_grid(
+                    sorted_table[key],
+                    uniques[key],
+                    heatmap_config["sensitivity"],
+                    heatmap_config["cutoff"],
+                )
+        else:
+            method_name = method_map[heatmap_settings.heatmap_type.value]
+            method = getattr(heatmap_core, method_name)
+            for key, heatmap_config in heatmap_settings.heatmap_config.items():
+                output[key] = method(sorted_table[key], uniques[key])
 
         return output
 
-    def quantile_classify(self, data):
+    def quantile_classify(self, calculations):
         quantile_index = {}
-        for key, a in data:
+        for key, a in calculations.items():
             quantile_index[key] = heatmap_core.quantile_classify(a)
         return quantile_index
 
+    def calculations_to_data_frame(self, uniques, calculations, quantiles):
+        dataframes = []
+        for oportunity, unique in uniques.items():
+            if not unique[0].size:
+                continue
+            # TODO: What to do whith empty arrays
+            start_time = time.time()
+            frame = pd.DataFrame(
+                data=[unique[0], calculations[oportunity], quantiles[oportunity]]
+            ).T
+            end_time = time.time()
+            print_info(f"converting {oportunity} time: {end_time-start_time} s")
+            frame.columns = ["id", oportunity, "class"]
+            dataframes.append(frame)
+
+        return dataframes
+
+    async def read_h3_dataframe(self, study_area_id, resolution):
+        base_path = "/app/src/cache/analyses_unit/"  # 9222/h3/10
+        directory = os.path.join(base_path, str(study_area_id), "h3")
+        file_name = os.path.join(directory, f"{resolution}.geojson")
+        dataframe = gpd.read_file(file_name)
```

Comment:
I assume this is very slow. We need to make sure to bypass geopandas. Or save the data directly as geopandas dataframe inside the file instead of geojson